CherkasovaDA/lab_3/main.cpp

```diff
@@ -0,0 +1,260 @@
+#include <iostream>
+
+using namespace std;
+
+struct MyType
+{
+    int a;
+};
+
+class MyVector
+{
+    int size = 0;
+    int capasity = 0;
+    MyType* massiv = 0;
+   
+    MyType NullItem()
+    {
+        MyType ItemNull{ 0 };
+        return ItemNull;
+    }
+    void PrintOneItem(int index)
+    {
+        cout << massiv[index].a;
+    }
+    void IncreasingTheCapasity()
+    {
+        MyType* massivCopy = new MyType[size];
+        for (int i = 0;i < size;i++)
+        {
+            massivCopy[i] = massiv[i];
+        }
+        capasity = capasity * 2;
+        massiv = new MyType[capasity];
+        for (int i = 0;i < size;i++)
+        {
+            massiv[i] = massivCopy[i];
+        }
+        for (int i = size + 1;i < capasity;i++)
+        {
+            massiv[i]=NullItem();
+        }
+        delete[]massivCopy;
+    }
+public:
+    MyVector()
+    {
+        size = 0;
+        capasity = 2;
+        massiv = new MyType[capasity];
```

Comment:
That is out of the scope of current task, but usually users of your class expects that default constructor does not allocate memory